scripts/gencert.service

```diff
@@ -6,8 +6,8 @@ After=forkexecd.service
 [Service]
 User=root
 Type=oneshot
-ExecStart=/opt/xensource/libexec/gencert /etc/xensource/xapi-pool-tls.pem xapi:pool
-ExecStart=/opt/xensource/libexec/gencert /etc/xensource/xapi-ssl.pem default
+ExecStart=/opt/xensource/libexec/gencert /etc/xensource/xapi-pool-tls.pem -1 xapi:pool
+ExecStart=/opt/xensource/libexec/gencert /etc/xensource/xapi-ssl.pem -1 default
```

Comment:
Won't this change of the argument list of /opt/xensource/libexec/gencert be wirde consequences?

By inserting the group id as 2nd argument before the (default|xapi:pool), this changes the API and affects all existing users, scripts, XenRT and other automation tools which update certificates? (e.g. internal and maybe by special customers which use automated ssh on Dom0 like the IBM Cloud)?

So far, I found these tools which have the string /opt/xensource/libexec/gencert in them, so it looks like they would be affected by this change:

```py
grep -r /opt/xensource/libexec/gencert /opt/xensource|cut -d' ' -f 3
/opt/xensource/debug/perftest
/opt/xensource/debug/event_listen
/opt/xensource/debug/vncproxy
/opt/xensource/debug/quicktestbin
/opt/xensource/bin/xe
/opt/xensource/bin/xapi-db-process
/opt/xensource/bin/xsh
/opt/xensource/bin/rrd2csv
/opt/xensource/bin/xapi
/opt/xensource/bin/mpathalert
/opt/xensource/libexec/v6d-citrix
/opt/xensource/libexec/alert-certificate-check
/opt/xensource/libexec/daily-license-check
/opt/xensource/libexec/gencert
/opt/xensource/libexec/xcp-rrdd-plugins/xcp-rrdd-squeezed
/opt/xensource/libexec/xcp-rrdd-plugins/xcp-rrdd-xenpm
/opt/xensource/libexec/xcp-rrdd-plugins/xcp-rrdd-gpumon
/opt/xensource/libexec/xcp-rrdd-plugins/xcp-rrdd-iostat
```

Reply:
> all existing users, scripts, XenRT and other automation tools which update certificates? 

If there are automation tools calling the script they are using the product in an unintended way. The only supported way to change the certificate is though xapi, the only intended user of gencert. If there is a usecase where the APIs are not enough we need to have a conversation with the clients so xapi can cater to it

> Also, couln't have this been done by a simple Exec of chmod 204 as a separate Exec call after the creation of the certificate? Then this change to Xapi could be avoided completely and only gencert.service would be updated.

Unfortunately, I don't think so. xapi calls the certificate library as well to bve able to install custom certificates (not just generate a new one), so it needs to be modified to know the gid as well. It is also used to control the pool-internal certificates

> So far, I found these tools which have the string /opt/xensource/libexec/gencert in them, so it looks like they would be affected by this change:

I believe most of them are false positives, they contain a library with this string, it doesn't mean they use it in a meaningful way